cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -160,6 +160,19 @@
 ;; TODO: strength reduction: div to shifts
 ;; TODO: div/rem by constants -> magic multiplications
 
+;; `(x >> k) << k` is the same as masking off the bottom `k` bits (regardless if
+;; this is a signed or unsigned shift right).
+(rule (simplify (ishl (fits_in_64 ty)
+                      (ushr ty x (iconst _ (u64_from_imm64 k)))
+                      (iconst _ (u64_from_imm64 k))))
+      (let ((mask u64 (u64_shl 0xFFFFFFFF k)))
```

Comment:
Should this be `0xFFFFFFFFFFFFFFFF` instead (i.e., `u64::MAX`)? Otherwise, a comment why `u32::MAX` is sufficient?

(Likewise below)

Reply:
D'oh!

Reply:
FWIW: I originally had `-1` but ISLE turns that into an `i128` literal in the rust source code and then it fails to type check.

Reply:
Yup, that's my fault. See #5423 for why I think it's better than the alternative, and #5431 for Chris' proposal for improving this situation in the future.